internal/api/service.go

```diff
@@ -2,37 +2,133 @@ package api
 
 import (
 	"context"
+	"errors"
 	"github.com/rs/zerolog/log"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 	"google.golang.org/protobuf/types/known/emptypb"
+	"ova-serial-api/internal/model"
+	"ova-serial-api/internal/repo"
 	api "ova-serial-api/pkg/ova-serial-api"
 )
 
 type OvaSerialAPI struct {
 	api.UnimplementedOvaSerialServer
+	repo repo.Repo
 }
 
-func NewSerialAPI() api.OvaSerialServer {
+func NewSerialAPI(repo repo.Repo) api.OvaSerialServer {
 	return &OvaSerialAPI{
-		UnimplementedOvaSerialServer: api.UnimplementedOvaSerialServer{},
+		repo: repo,
 	}
 }
 
 func (a *OvaSerialAPI) CreateSerialV1(ctx context.Context, req *api.CreateSerialRequestV1) (*api.CreateSerialResponseV1, error) {
-	log.Info().Msgf("CreateSerial request: %v", req)
-	return nil, nil
+	if err := req.Validate(); err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	log.Debug().
+		Int64("UserId", req.UserId).
+		Str("Title", req.Title).
+		Str("Genre", req.Genre).
+		Uint32("Year", req.Year).
+		Uint32("Seasons", req.Seasons).
+		Msg("Create serial")
```

Comment:
Честно говоря, не знаю, можно ли так в этой библиотеке, но здесь должно быть что-то вроде этого:
```suggestion
	log.Debugf("Create serial: %+v", req)
```
Правила составления этого шаблона – как в `fmt.Sprintf()`

Reply:
Нужно просто найти метод нужный